extensions/kafka-eight-simpleConsumer/src/main/java/io/druid/firehose/kafka/KafkaEightSimpleConsumerFirehoseFactory.java

```diff
@@ -0,0 +1,318 @@
+      private volatile InputRow row = null;
+
+      {
+        lastOffsetPartitions = Maps.newConcurrentMap();
+        lastOffsetPartitions.putAll(lastOffsets);
+      }
+
+      @Override
+      public void start() throws Exception
+      {
+        nextMessage();
+      }
+
+      @Override
+      public boolean advance()
+      {
+        if (stop) {
+          return false;
+        }
+
+        nextMessage();
+        return true;
+      }
+
+      private void nextMessage()
+      {
+        try {
+          row = null;
+          while (row == null) {
+            if (msg != null) {
+              lastOffsetPartitions.put(msg.getPartition(), msg.offset());
+            }
+
+            msg = messageQueue.take();
+            interrupted = false;
+
+            final byte[] message = msg.message();
+            row = message == null ? null : theParser.parse(ByteBuffer.wrap(message));
+          }
+        }
+        catch (InterruptedException e) {
+          interrupted = true;
+          log.info(e, "Interrupted when taken from queue");
+        }
+      }
+
+      @Override
+      public InputRow currRow()
+      {
+        if (interrupted) {
+          return null;
+        }
+        return row;
+      }
+
+      @Override
+      public Committer makeCommitter()
+      {
+        final Map<Integer, Long> offsets = Maps.newHashMap(lastOffsetPartitions);
+
+        return new Committer()
+        {
+          @Override
+          public Object getMetadata()
+          {
+            return offsets;
+          }
+
+          @Override
+          public void run()
+          {
+
+          }
+        };
+      }
+
+      @Override
+      public void close() throws IOException
+      {
+        log.info("Stopping kafka 0.8 simple firehose");
+        stop = true;
+        for (PartitionConsumerWorker t : consumerWorkers) {
+          t.close();
+        }
+      }
+    };
+  }
+
+  private static class PartitionConsumerWorker implements Closeable
+  {
+    private final String topic;
+    private final KafkaSimpleConsumer consumer;
+    private final int partitionId;
+    private final long startOffset;
+
+    private final AtomicBoolean stopped = new AtomicBoolean(false);
+    private volatile Thread thread = null;
+
+    PartitionConsumerWorker(String topic, KafkaSimpleConsumer consumer, int partitionId, long startOffset)
+    {
+      this.topic = topic;
+      this.consumer = consumer;
+      this.partitionId = partitionId;
+      this.startOffset = startOffset;
+    }
+
+    public void go(final LinkedBlockingQueue<BytesMessageWithOffset> messageQueue) {
+      thread = new Thread()
+      {
+        @Override
+        public void run()
+        {
+          long offset = startOffset;
+          log.info("Start running parition[%s], offset[%s]", partitionId, offset);
+          try {
+            while (!stopped.get()) {
+              try {
+                Iterable<BytesMessageWithOffset> msgs = consumer.fetch(offset, 10000);
```

Comment:
what does this number 10000 meant for, can we make this a constant with a meaningful name ?